library/src/main/java/com/bumptech/glide/load/resource/gif/GifDrawable.java

```diff
@@ -390,6 +407,50 @@ public void setLoopCount(int loopCount) {
     }
   }
 
+  /**
+   * Register callback to listen to GifDrawable animation end event after specific loop count
+   * set by {@link GifDrawable#setLoopCount(int)}
+   *
+   * Note: Unregister this in onLoadCleared to avoid potential memory leak.
+   * @see GifDrawable#unregisterAnimationCallback(AnimationCallback).
+   *
+   * @param animationCallback Animation callback {@link Animatable2Compat.AnimationCallback}.
+   */
+  @Override
+  public void registerAnimationCallback(@NonNull AnimationCallback animationCallback) {
+    if (animationCallback == null) {
```

Comment:
I'd either skip this check or make it a Precondition. I think with the `@NonNull` annotation you're fine just skipping the check.

If you do want a check, I'd prefer a Precondition because it would immediately show the developer where they went wrong, where the existing check will just silently fail.